### PR TITLE
Allow instance instantiation with private constructor

### PR DESCRIPTION
**Pull request by tothegills**
Relates to OctopusDeploy/Issues#5325

### CODE REVIEW

**Comment by tothegills** [comment]
[Tests demonstrating reporting endpoints work #3496](https://trello.com/c/iFyk8N6U/192-tests-demonstrating-reporting-endpoints-work-3496)

File: source/Nevermore/RelationalTransaction.cs [hunk 1]
```diff
@@ -461,7 +461,7 @@ IEnumerable<T> Stream<T>(IDbCommand command, DocumentMap mapping)
                         }
                         else
                         {
-                            instance = (T) Activator.CreateInstance(instanceType);
+                            instance = (T) Activator.CreateInstance(instanceType, true);
```
Comment (johnsimons) [inline source/Nevermore/RelationalTransaction.cs]: @tothegills could this be considered a breaking change?

Comment (tothegills) [reply source/Nevermore/RelationalTransaction.cs]: Yes I agree, there could be private constructors that are now preferred for instance creation.

Comment (johnsimons) [reply source/Nevermore/RelationalTransaction.cs]: So, how do we move forward, should this be a flag in Nevermore that by default if off?

**Pull request merged**